ts/Data/Stores/CSVDataStore.ts

```diff
@@ -0,0 +1,225 @@
+    private liveDataURL?: string;
+    private liveDataTimeout?: number;
+
+    private addEvents(): void {
+        this.on('load', (e: CSVDataStore.LoadEventObject): void => {
+            this.dataParser.parse({
+                csv: e.csv,
+                ...this.parserOptions
+            });
+            if (this.liveDataURL) {
+                this.poll();
+            }
+            fireEvent(this, 'afterLoad', { table: this.dataParser.getTable() });
+        });
+        this.on('afterLoad', (e: DataStore.LoadEventObject): void => {
+            this.table = e.table;
+        });
+        this.on('parse', (e: DataStore.ParseEventObject): void => {
+            // console.log(e)
+        });
+        this.on('fail', (e: any): void => {
+            // throw new Error(e.error)
+        });
+    }
+
+
+    /**
+     * Handle polling of live data
+     */
+    private poll(): void {
+        const { dataRefreshRate, enablePolling: pollingEnabled, csvURL } = this.options;
+        const updateIntervalMs = (dataRefreshRate > 1 ? dataRefreshRate : 1) * 1000;
+        if (pollingEnabled && csvURL === this.liveDataURL) {
+            // We need to stop doing this if the URL has changed
+            this.liveDataTimeout = setTimeout((): void => {
+                this.fetchCSV();
+            }, updateIntervalMs);
+        }
+    }
+
+    private fetchCSV(initialFetch?: boolean): void {
+        const store = this,
+            maxRetries = 3,
+            { csvURL } = store.options;
+        let currentRetries: number;
+
+        if (initialFetch) {
+            clearTimeout(store.liveDataTimeout);
+            store.liveDataURL = csvURL;
+        }
+
+        ajax({
+            url: store.liveDataURL,
+            dataType: 'text',
+            success: function (csv: string): void {
+                fireEvent(store, 'load', { csv });
+            },
+            error: function (xhr, text): void {
+                if (++currentRetries < maxRetries) {
+                    store.poll();
+                }
+                fireEvent(store, 'fail', { error: { xhr, text } });
+            }
+        });
+    }
+
+    public load(): void {
+        const store = this,
+            { csv, csvURL } = store.options;
+        if (csv) {
+            fireEvent(store, 'load', { csv });
+        } else if (csvURL) {
+            store.fetchCSV(true);
+        }
+    }
+
+    public save(): void {
+
+    }
+
+    public toJSON(): CSVDataStore.ClassJSON {
+        const json: CSVDataStore.ClassJSON = {
+            $class: 'CSVDataStore',
+            options: merge(this.options),
+            table: this.table.toJSON(),
+            metadata: []
+        };
+
+        let metadata;
+
+        for (let i = 0, iEnd = this.metadata.length; i < iEnd; i++) {
+            metadata = this.metadata[i];
+
+            json.metadata.push([
+                metadata.name,
+                metadata.metadata
+            ]);
+        }
```

Comment:
@wchmiel 
This code can be found in all store implementations and therefor should go into the `DataStore` class as a `protected getMetadataJSON(): DataStore.MetadataJSON` function.

Reply:
We just discuss it and will refactor ;) 

Thanks!